encoder/encoder.cpp

```diff
@@ -16,9 +17,28 @@
 #include "libav_encoder.hpp"
 #endif
 
-Encoder *Encoder::Create(VideoOptions const *options, const StreamInfo &info)
+Encoder *auto_codec_select(VideoOptions *options, const StreamInfo &info)
 {
-	if (strcasecmp(options->codec.c_str(), "yuv420") == 0)
+	const char hw_codec[] = "/dev/video11";
+	struct stat buffer;
+
+	if (!stat(hw_codec, &buffer))
+		return new H264Encoder(options, info);
+
+#if LIBAV_PRESENT
+	// No hardware codec available, use x264 through libav.
+	options->libav_video_codec = "libx264";
+	return new LibAvEncoder(options, info);
+#endif
+
+	return new MjpegEncoder(options);
+}
```

Comment:
Is the libav codec available by default these days? I'm not sure about falling back to MJPEG just because you don't have a /dev/video11?

Reply:
On the full fat builds it is.  Not sure what the best fallback without HW codec or libav would be?  Perhaps plain YUV?